tests/specs/file.js

```diff
@@ -134,363 +195,576 @@ define(["dist/amd-resolver"], function(Resolver) {
-          expect(urlo.protocolmark).to.equal("//");
-          expect(urlo.hostname).to.equal("hoistedjs.com");
-          expect(urlo.port).to.be.an('undefined');
-          expect(urlo.path).to.equal("/");
-          expect(urlo.search).to.equal("?topic=data");
-          expect(urlo.hash).to.equal("#migration/path");
-        });
-
-        it("url with search and hash and simple path", function() {
-          var urlo = File.parseUri("http://hoistedjs.com/moretesting?topic=data#migration/path");
-          expect(urlo.origin).to.equal("http://hoistedjs.com");
-          expect(urlo.protocol).to.equal("http:");
-          expect(urlo.protocolmark).to.equal("//");
-          expect(urlo.hostname).to.equal("hoistedjs.com");
-          expect(urlo.port).to.be.an('undefined');
-          expect(urlo.path).to.equal("/moretesting");
-          expect(urlo.search).to.equal("?topic=data");
-          expect(urlo.hash).to.equal("#migration/path");
-        });
-
-        it("url with search and hash and simple path with port", function() {
-          var urlo = File.parseUri("http://hoistedjs.com:599/moretesting?topic=data#migration/path");
-          expect(urlo.origin).to.equal("http://hoistedjs.com:599");
-          expect(urlo.protocol).to.equal("http:");
-          expect(urlo.protocolmark).to.equal("//");
-          expect(urlo.hostname).to.equal("hoistedjs.com");
-          expect(urlo.port).to.equal('599');
-          expect(urlo.path).to.equal("/moretesting");
-          expect(urlo.search).to.equal("?topic=data");
-          expect(urlo.hash).to.equal("#migration/path");
-        });
-      });
-
-
-      describe("FILE", function() {
-        it("Empty string", function() {
-          var urlo, _ex;
-          try {
-            urlo = File.parseUri("");
-          }
-          catch (ex) {
-            _ex = ex;
-            expect(ex.message).to.equal("Must provide a string to parse");
-          }
-          finally {
-            expect(_ex).to.be.an('object');
-            expect(urlo).to.be.an('undefined');
-          }
-        });
-
-        it("Single forward slash", function() {
-          var urlo = File.parseUri("/");
-          expect(urlo.origin).to.be.an('undefined');
-          expect(urlo.protocol).to.be.an('undefined');
-          expect(urlo.protocolmark).to.be.an('undefined');
-          expect(urlo.hostname).to.be.an('undefined');
-          expect(urlo.port).to.be.an('undefined');
-          expect(urlo.path).to.equal("/");
-          expect(urlo.search).to.be.an('undefined');
-          expect(urlo.hash).to.be.an('undefined');
+    describe("when calling parseUri", function() {
+      describe("using HTTP protocol", function() {
+        describe("and a simple url", function() {
+          describe("for `http://hoistedjs.com`", function() {
+            var parsedURI = File.parseUri("http://hoistedjs.com");
+
+            it("origin should equal `http://hoistedjs.com`", function() {
```

Comment:
@Mark-Simulacrum all the tests that have the `should`s would read better without them, and would also be more accurate.  There are assertion checks you can do that are `should` instead of `expect`.  So that can be misleading when reading the unit test reports.